api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -30,6 +30,15 @@
  */
 public interface Table {
 
+  /**
+   * Return the full name for this table.
+   *
+   * @return this table's name
+   */
+  default String name() {
```

Comment:
It is debatable whether we have to default this. I did this to avoid breaking custom implementations.